fix(test): use app path directly without looping over bench path

blaggacao opened the pull request:
> Please provide enough information so that others can review your pull request:

The bench path heuristic is flawed when apps are installed as python modules.

Forthemore the current code is an unnecessary indirection (over that flawed code path).

> Explain the **details** for making this change. What existing problem does the pull request solve?

By avoiding this unnecessary indirection, these particular code paths are fixed in those scenarios.

It also simplifies the implementation a bit.

> Screenshots/GIFs

<!-- Add images/recordings to better visualize the change: expected/current behviour -->

cc @teutat3s

blaggacao commented:
ping

blaggacao commented:
@ankush This helps green tests that depend on a [test python env](https://github.com/blaggacao/frappix/blob/3adc1361940278496aa982074e29edfa7d8d274f/src/tests.nix#L52) that not necessarily depends on a particular folder layout. It otherwise does not modify any current behaviour and makes the code more immediate.

barredterra commented on frappe/tests/test_patches.py:
This file now has an unused `import get_bench_path`

blaggacao replied:
Addressed in [force-push](https://github.com/frappe/frappe/compare/9d075afbb92a9b5916d28f00c2dc0145391a90c6..fb11f9a8fa275a0b6793e75f86871a4f6bef7bb1)

barredterra commented on frappe/tests/test_translate.py:
```diff
@@ -45,7 +45,9 @@ def tearDown(self):
 		data = frappe.translate.get_messages_from_file(translation_string_file)
-		exp_filename = "apps/frappe/frappe/tests/translation_test_file.txt"
+		bench_path = get_bench_path()
+		file_path = frappe.get_app_path("frappe", "tests/translation_test_file.txt")
```
```suggestion
		file_path = frappe.get_app_path("frappe", "tests", "translation_test_file.txt")
```

I guess we won't run on windows any time soon, but why presume to know the path separator if we can avoid it...

blaggacao replied:
Incorporated in [force-push](https://github.com/frappe/frappe/compare/9d075afbb92a9b5916d28f00c2dc0145391a90c6..fb11f9a8fa275a0b6793e75f86871a4f6bef7bb1)

barredterra commented on frappe/tests/test_utils.py:
```diff
@@ -422,9 +422,9 @@ def test_validate_name(self):
 class TestImage(FrappeTestCase):
 	def test_strip_exif_data(self):
-		original_image = Image.open("../apps/frappe/frappe/tests/data/exif_sample_image.jpg")
+		original_image = Image.open(frappe.get_app_path("frappe", "tests/data/exif_sample_image.jpg"))
```
```suggestion
		original_image = Image.open(frappe.get_app_path("frappe", "tests", "data", "exif_sample_image.jpg"))
```

blaggacao replied:
Incorporated in [force-push](https://github.com/frappe/frappe/compare/9d075afbb92a9b5916d28f00c2dc0145391a90c6..fb11f9a8fa275a0b6793e75f86871a4f6bef7bb1)

barredterra commented on frappe/tests/test_utils.py:
```diff
@@ -422,9 +422,9 @@ def test_validate_name(self):
+		original_image = Image.open(frappe.get_app_path("frappe", "tests/data/exif_sample_image.jpg"))
 		original_image_content = open(
-			"../apps/frappe/frappe/tests/data/exif_sample_image.jpg", mode="rb"
+			frappe.get_app_path("frappe", "tests/data/exif_sample_image.jpg"), mode="rb"
```
```suggestion
			frappe.get_app_path("frappe", "tests", "data", "exif_sample_image.jpg"), mode="rb"
```

blaggacao replied:
Incorporated in [force-push](https://github.com/frappe/frappe/compare/9d075afbb92a9b5916d28f00c2dc0145391a90c6..fb11f9a8fa275a0b6793e75f86871a4f6bef7bb1)

barredterra commented on frappe/tests/test_utils.py:
```diff
@@ -434,7 +434,7 @@ def test_strip_exif_data(self):
 
 	def test_optimize_image(self):
-		image_file_path = "../apps/frappe/frappe/tests/data/sample_image_for_optimization.jpg"
+		image_file_path = frappe.get_app_path("frappe", "tests/data/sample_image_for_optimization.jpg")
```
```suggestion
		image_file_path = frappe.get_app_path("frappe", "tests", "data", "sample_image_for_optimization.jpg")
```

blaggacao replied:
Incorporated in [force-push](https://github.com/frappe/frappe/compare/9d075afbb92a9b5916d28f00c2dc0145391a90c6..fb11f9a8fa275a0b6793e75f86871a4f6bef7bb1)

barredterra commented on frappe/website/doctype/website_theme/website_theme.py:
This file now has an unused `import get_path`

blaggacao replied:
Addressed in [force-push](https://github.com/frappe/frappe/compare/9d075afbb92a9b5916d28f00c2dc0145391a90c6..fb11f9a8fa275a0b6793e75f86871a4f6bef7bb1)

barredterra reviewed (commented):
Some nitpicks, functionality looks fine 👍🏼

blaggacao committed:
fix(test): ensure tests construe the site url from its constructor

blaggacao commented:
@barredterra Do you think this is ready? I think (hope) that I've addressed all of your feedback. Thanks again for your kind review!!

I can remove some [patches](https://github.com/blaggacao/frappix/blob/3adc1361940278496aa982074e29edfa7d8d274f/src/overlays/frappe/default.nix#L57-L61) in [frappix](https://github.com/blaggacao/frappix) after this.

barredterra commented:
@blaggacao your force-push makes it hard to check your recent changes without reviewing everything from scratch again. 😕

blaggacao commented:
Ok, sorry, noted. @barredterra 

Thank you for merging!
